text/0000-dyn-error-generic-member-access.md

```diff
@@ -228,24 +226,31 @@ Here is the implementation for the proof of concept, taken from Nika Layzell's
 [object-provider crate]:
 
 ```rust
+use core::any::TypeId;
+use core::cell::Cell;
+use core::fmt;
+use core::marker::PhantomData;
+use core::pin::Pin;
+
 /// A dynamic request for an object based on its type.
 ///
-/// `'r` is the lifetime of request, and `'out` is the lifetime of the requested
-/// reference.
-pub struct Request<'r, 'out> {
-    buf: NonNull<TypeId>,
-    _marker: PhantomData<&'r mut &'out Cell<()>>,
-}
-
-impl<'r, 'out> Request<'r, 'out> {
+/// `'out` is the lifetime of the requested reference.
+#[repr(transparent)]
+pub struct Request<'out>(RequestBuf<PhantomData<&'out Cell<()>>>);
```

Comment:
need to include a `PhantomPinned` somewhere (maybe inside `RequestBuf` for simplicity), to avoid this type becoming `Unpin`.